hkmc2/shared/src/main/scala/hkmc2/codegen/HandlerLowering.scala

```diff
@@ -393,6 +393,9 @@ class HandlerLowering(paths: HandlerPaths)(using TL, Raise, Elaborator.State, El
           ResultPlaceholder(res, freshId(), c2, k(Value.Ref(res)))
         case r => super.applyResult2(r)(k)
       override def applyLam(lam: Value.Lam): Value.Lam =
+        // This should normally be unreachable, but we can just emit a warning and proceed with the transformation
+        raise(WarningReport(msg"BUG: Unexpected lambda during handler lowering" -> lam.toLoc :: Nil,
```

Comment:
The correct thing to report here is an
```suggestion
        raise(InternalError(msg"Unexpected lambda during handler lowering" -> lam.toLoc :: Nil,
```